src/main/java/com/couchbase/lite/storage/SQLiteStorageEngineBase.java

```diff
@@ -72,12 +72,20 @@ public boolean open(String path, SymmetricKey encryptionKey) throws SQLException
                     getWALConnectionPoolSize(),
                     null, new ConnectionListener());
             Log.v(Log.TAG_DATABASE, "%s: Opened Android sqlite db", this);
-        } catch(SQLiteDatabaseCorruptException e) {
+        } catch (SQLiteDatabaseCorruptException e) {
             hasError = true;
-            Log.e(Log.TAG_DATABASE, "Unauthorized to open the SQLite database", e);
-            throw new SQLException(SQLException.SQLITE_ENCRYPTION_UNAUTHORIZED,
-                    "Cannot decrypt or access the database");
-        } catch(com.couchbase.lite.internal.database.SQLException e) {
+            // SQLCipher with encryption
+            if (encryptionKey != null && supportEncryption()) {
```

Comment:
This check doesn't cover 2 things:
1. If database is encrypted but the library used is sqlite instead of sqlcipher.
2. If database is encrypted but no encryption key given.